python3.8/cookiecutter-aws-sam-efs-python/{{cookiecutter.project_name}}/tests/integration/test_api_gateway.py

```diff
@@ -0,0 +1,33 @@
+import os
+from unittest import TestCase
+
+import boto3
+import requests
+
+
+class TestApiGateway(TestCase):
+    api_endpoint: str
+
+    def setUp(self) -> None:
+        """
+        Based on the provided env variable AWS_SAM_STACK_NAME,
+        here we use cloudformation API to find out what the ServerlessRestApi URL is
+        """
+        stack_name = os.environ.get("AWS_SAM_STACK_NAME")
+        self.failIf(not stack_name, "Cannot find env var AWS_SAM_STACK_NAME")
+
+        client = boto3.client("cloudformation")
+        response = client.list_stack_resources(StackName=stack_name)
+        resources = response["StackResourceSummaries"]
+        api_resources = [resource for resource in resources if resource["LogicalResourceId"] == "ServerlessRestApi"]
+        self.failIf(not api_resources, "Cannot find ServerlessRestApi")
+
+        self.api_endpoint = f"https://{api_resources[0]['PhysicalResourceId']}.execute-api.{client.meta.region_name}.amazonaws.com/Prod/hello/"
```

Comment:
Can we get the endpoint from stack output, instead of this way?

Reply:
good idea, changed